src/internal/_getTypedFunction.js

```diff
@@ -0,0 +1,6 @@
+var _FN_TYPE_KEY = require('./_FN_TYPE_KEY');
+
+module.exports = function _getTypedFunction(type, f) {
+    f[_FN_TYPE_KEY] = type;
+    return f;
+};
```

Comment:
dunno! 

also, i think if we are going to start adding metadata to functions we should have a plan for it, and not just proceed in an _ad hoc_ fashion. And I do think we are going to start exploring the use of metadata on funcitons, probably in a `.ramda` namespace. I have an idea I wanna try out ....